File: src/it/gtd/freetether/UnlockController.scala
package it.gtd.freetether

import android.content.{Intent, Context, BroadcastReceiver}
import org.scaloid.common.{Logger, TagUtil}

class UnlockController extends BroadcastReceiver with Logger with TagUtil {
  def onReceive(context: Context, intent: Intent) = new WifiAP(context).state match {
    case Enabled =>
      info(s"$Enabled received")
      Tether.unlock()
    case Disabled => Tether.lock()
      info(s"$Disabled received")
    case x =>
      info(s"$x received")
  }
}

File: src/main/java/it/gtd/freetether/UnlockController.java
package it.gtd.freetether;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import it.gtd.freetether.wifi.AccessPoint;
import it.gtd.freetether.wifi.Tether;

public final class UnlockController extends BroadcastReceiver {
    public void onReceive(Context context, Intent intent) {
        switch (AccessPoint.of(context).getState()) {
            case Enabled:
                Tether.getInstance().unlock();
                break;
            case Disabled:
                Tether.getInstance().lock();
                break;
        }
    }
}
